babel/messages/extract.py

```diff
@@ -398,7 +398,13 @@ def extract_python(fileobj, keywords, comment_tags, options):
     in_def = in_translator_comments = False
     comment_tag = None
 
-    encoding = parse_encoding(fileobj) or options.get('encoding', 'iso-8859-1')
+    # In Python2, the default source encoding is US-ASCII
+    # In Python3, the default source encoding is UTF-8
+    # http://legacy.python.org/dev/peps/pep-0008/#source-file-encoding
+    default_encoding = 'US-ASCII' if PY2 else 'UTF-8'
```

Comment:
If the file is ascii it will succeed in both python 2 and python 3. Latin1 is a superset of ascii.

If a file has a latin1 encoded character and does not have an encoding pragma it will be a syntax error in both python2 or python3 (without even factoring in babel)

If the file has a glyph that is latin1 representable (such as the one presented) that is encoded as utf8 and does not have an encoding pragma:
- python2: syntax error
- python3: this is fine (the default source encoding is utf8)